custom_components/ytube_music_player/select.py

```diff
@@ -0,0 +1,124 @@
+"""Platform for sensor integration."""
+import logging
+from homeassistant.components.select import SelectEntity
+from homeassistant.exceptions import NoEntitySpecifiedError
+from . import DOMAIN
+from .const import *
+
+_LOGGER = logging.getLogger(__name__)
+
+async def async_setup_entry(hass, config, async_add_entities):
+	entities = []
+	entities.append(yTubeMusicPlaylistSelect(hass, config))
+	entities.append(yTubeMusicSpeakerSelect(hass, config))
+	entities.append(yTubeMusicPlayModeSelect(hass, config))
+	entities.append(yTubeMusicSourceSelect(hass, config))
+	entities.append(yTubeMusicRepeatSelect(hass, config))
+	async_add_entities(entities, update_before_add=True)
+
+class yTubeMusicSelectEntity(SelectEntity):
+	def __init__(self, hass, config):
+		self.hass = hass
+		self._device_id = config.entry_id
+		self._device_name = config.data.get(CONF_NAME)
+		self._attr_has_entity_name = True
+
+	def select_option(self, option):
+		"""Change the selected option."""
+		self._attr_current_option = option
+		self.async_write_ha_state()
+
+	@property
+	def device_info(self):
+		return {
+			'identifiers': {(DOMAIN, self._device_id)},
+			'name': self._device_name,
+			'manufacturer': "Google Inc.",
+			'model': DOMAIN
+		}
+
+	@property
+	def should_poll(self):
+		return False
+
+
+class yTubeMusicPlaylistSelect(yTubeMusicSelectEntity):
+	def __init__(self, hass, config):
+		super().__init__(hass, config)
+		self._attr_unique_id = self._device_id + "_playlist"
+		self._attr_name = "Playlist"
+		self._attr_icon = 'mdi:playlist-music'
+		self.hass.data[DOMAIN][self._device_id]['select_playlist'] = self
+		self._attr_options = ["loading"]
+		self._attr_current_option = None
+
+	async def async_update(self):
+		# update select
+		self._ready = True
```

Comment:
what is the purpose of the _ready? is HA reading them?

Reply:
Sorry,I just copied them from your sensor.py.I'm not sure what this is used for either.